### ISSUE TITLE
Duplicate keys in `:alpha`betic style

### ISSUE THREAD

**Issue by kellertuer** [JuliaDocs/DocumenterCitations.jl#26]
Consider my (little lazy) example

```
@techreport{deCasteljau:1959,
    AUTHOR      = {de Casteljau, Paul},
    institution   = {Enveloppe Soleau 40.040, Institute National de la Propriété Industrielle, Paris.},
    TITLE       = {Outillage methodes calcul},
    YEAR        = {1959}
}
@techreport{deCasteljau:1959b,
    AUTHOR      = {de Casteljau, Paul},
    INSTITUTION = {Enveloppe Soleau 40.040, Institute National de la Propriété Industrielle, Paris.},
    TITLE       = {Outillage methodes calcul - my lazy b},
    YEAR        = {1959}
}
```

and a `temp.md`.

````
[deCasteljau:1959](@cite), [deCasteljau:1959b](@cite)
```@bibliography
Pages = ["temp.md"]
Canonical=false
```
````

Then the resulting page looks like

<img width="836" alt="Screenshot 2023-07-17 at 17 04 05" src="https://github.com/JuliaDocs/DocumenterCitations.jl/assets/115566/4253ae34-272b-4d24-b232-3771c4735007">

So the generated _key_ clashes (considering them like hashes). They should at least be `[Cas59a]`, and `[Cas59b]` I think?

I might be mistaken, but I think the Keys would also respect the prefix of the name so that it would actually be `[dCa69]` – but maybe I remember that wrongly as well. The main part of this issue is the clash in keys, since this might also happen for other “Key collisions” I think (same author last names first letters in the same year for example).

PS: I promise this is the last issue for now, I even managed to rework my references and have a PR at https://manoptjl.org/previews/PR281/ that is (besides checking that non canonical bibliographies exist on all pages with references) reworked and I think it is overall quite nice! I just stumbled about this issue here by accident, so it does not appear in my references.

**Comment by goerz** [comment]
You are absolutely correct! I thought I could slide this one by for the moment 😉

The way I was planning to fix this was to have a slightly smarter `:alpha` style. More specifically, one would have an `AlphaStyle` object that gets passed as `style` to `CitationBibliography`, and which would keep track of all generated labels, and modify them appropriately when there are duplications. Similar to (but simpler than) the approach in https://github.com/JuliaDocs/DocumenterCitations.jl/issues/13, and in fact a good example for a "stateful" style to be discussed in the documentation.

I'd prioritize this a bit more if someone actually wanted to use the `:alpha` style "in the wild", but it will be fixed eventually, in any case. (Just probably not in the next few weeks, due to conference travel.)

> I promise this is the last issue for now

Not at all! Keep them coming!

> I even managed to rework my references and have a [PR](https://manoptjl.org/previews/PR281/) at  that is [...] reworked and I think it is overall quite nice!

When it's merged and the documentation is online, feel free to make a PR here to add Manopt to the [list of examples](https://juliadocs.org/DocumenterCitations.jl/dev/#Examples).

**Comment by kellertuer** [comment]
I will use the alpha style “in the wild”! But no worries, for now I do not have such a collision and this is also really not urgent.

Yes once I have finished the PR (maybe this week), I will add it to the list of examples, for sure.

**Comment by lgoettgens** [comment]
I am trying to convert [Oscar.jl](https://github.com/oscar-system/Oscar.jl) to the `v1.0.0` and would like to stay with the `:alpha` mode that was decided on long ago. Unfortunately, there occur some collisions of the above form.
I think this is a blocker for us to use it until this is fixed.

**Comment by goerz** [comment]
Alright, I'll give it some priority

**Comment by lgoettgens** [comment]
We change to https://juliadocs.org/DocumenterCitations.jl/v1.0.0/gallery/#Custom-style:-Citation-key-labels for the time being, as we can put working and distinct labels in the `.bib` file.

**Comment by goerz** [comment]
That’s definitely a workaround! It also has the benefit that you have the same labels in the docstrings as in the rendered documentation, when reading them in a text editor.

I might be able to push out a smarter `:alpha` style today, though.

**Comment by goerz** [comment]
Should be fixed in #31. Have a look if this works as expected, if you can

**Comment by lgoettgens** [comment]
> Should be fixed in #31. Have a look if this works as expected, if you can

I unfortunately will be out of office the whole week, and without any device running julia.

**Comment by goerz** [comment]
I'm okay with waiting a little bit…

**Comment by kellertuer** [comment]
Thanks for working on that, due to conferences and travels, II will only have time around Wednesday to take a look, but if you are sure it works merge it without my feedback for sure.

**Comment by goerz** [comment]
Take your time, this is really for you guys' benefit, so it makes more sense to wait until you've had a chance to look at it 😉

**Comment by lgoettgens** [comment]
> Should be fixed in #31. Have a look if this works as expected, if you can

The preview in https://juliadocs.org/DocumenterCitations.jl/previews/PR31/gallery/#alphabetic_style looks great. Unfortunately, I cannot get it to work on my repo, but that may be due to my local setup.
